reference-implementation/lib/helpers.js

```diff
@@ -101,7 +101,19 @@ exports.PromiseInvokeOrPerformFallback = (O, P, args, F, argsF) => {
 };
 
 // Not implemented correctly
-exports.TransferArrayBuffer = O => O.slice();
+exports.TransferArrayBuffer = O => {
+  const transferredIshVersion = O.slice();
+
+  // This is specifically to fool tests that test "is transferred" by taking a non-zero-length
+  // ArrayBuffer and checking if its byteLength starts returning 0.
+  Object.defineProperty(O, "byteLength", {
+    get() {
+      return 0;
+    }
+  });
```

Comment:
I think I'm a bit in awe with this hack :)

Reply:
Wait until you see what I'm going to tack onto this PR for solving #751...